### PR TITLE
mainnet native staking fork tests

### PR DESCRIPTION
**Pull request by naddison36**
# Contract changes

* Fail accounting if activeDepositedValidators < fullyWithdrawnValidators
* Changed Harvester to transfer WETH to Dripper
* Removed unused imports

# Other changes

* Deploy script now sets validator registrator and upgrades the OETH Harvester
* mainnet fork tests
* Holesky re-deployment: https://github.com/OriginProtocol/origin-dollar/pull/2046

### CODE REVIEW

File: contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol [hunk 1]
```diff
@@ -164,13 +161,21 @@ abstract contract ValidatorAccountant is ValidatorRegistrator {
...
-                _pause();
-            }
-            // fail the accounting
-            accountingValid = false;
+            return _failAccounting(pauseOnFail);
+        }
+    }
+
+    /// @dev pause any further accounting if required and return false
+    function _failAccounting(bool pauseOnFail)
```
Comment (sparrowDom) [inline contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol:169]: nice cleanup

File: contracts/deploy/mainnet/091_native_ssv_staking.js [hunk 2]
```diff
@@ -114,11 +114,23 @@ module.exports = deploymentWithGovernanceProposal(
     // 7. Safe approve SSV token spending
     await cStrategy.connect(sDeployer).safeApproveAllTokens();
 
+    // 8. Deploy Harvester
+    const cOETHHarvesterProxy = await ethers.getContract("OETHHarvesterProxy");
```
Comment (sparrowDom) [inline contracts/deploy/mainnet/091_native_ssv_staking.js:118]: thanks for catching this

File: contracts/utils/addresses.js [hunk 3]
```diff
@@ -245,7 +245,15 @@ addresses.mainnet.CurveCVXPool = "0xB576491F1E6e5E62f1d8F26062Ee822B40B0E0d4";
 addresses.mainnet.SSV = "0x9D65fF81a3c488d585bBfb0Bfe3c7707c7917f54";
 addresses.mainnet.SSVNetwork = "0xDD9BC35aE942eF0cFa76930954a156B3fF30a4E1";
 addresses.mainnet.beaconChainDepositContract =
-  "0x00000000219ab540356cbb839cbe05303d7705fa";
+  "0x00000000219ab540356cBB839Cbe05303d7705Fa";
+
+// Native Staking Strategy
+addresses.mainnet.nativeStakingSSVStrategyProxy =
+  "0xdF58F78cebbb2A60740eD2f86cDf0545a485102F";
+addresses.mainnet.nativeStakingFeeAccumulatorProxy =
```
Comment (sparrowDom) [inline contracts/utils/addresses.js:253]: The nativeStakingFeeAccumulator and SSV Strategy proxy seem to not be deployed to the above addresses or anywhere for the moment

**Review by sparrowDom — COMMENTED** [review]
LGTM 
there are a couple of nitpicks let me sort those out

File: contracts/utils/addresses.js (hunk 3 above)
Comment (sparrowDom) [reply contracts/utils/addresses.js:253]: I will just comment them out for the moment

File: contracts/utils/resolvers.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  // if put outside this function, the following error occurs:
+  // "Hardhat can't be initialized while its config is being defined"
+  const hre = require("hardhat");
+  const isFork = process.env.FORK === "true";
+  const isMainnet = hre.network.name === "mainnet";
+  const isMainnetOrFork = isMainnet || isFork;
+
+  if (isMainnetOrFork) {
+    const assetAddr =
+      addresses.mainnet[symbol] || addresses.mainnet[symbol + "Proxy"];
```
[the remark this answers is not in the document]
Comment (sparrowDom) [reply contracts/utils/resolvers.js:21]: I'll turn those 2 around

**Comment by sparrowDom** [comment]
@naddison36 I've:
- consolidated the fork tests on Mainnet and Holesky to share the behaviour tests like code
- corrected the holesky deploy script that was missing re-deployment of the Harvester and configuration of the harvester. It would be cleaner to have that as a separate PR, but for convenience of getting tests to pass I've put it into this one

File: contracts/utils/resolvers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (naddison36) [reply contracts/utils/resolvers.js]: good pickup. Thanks

File: contracts/utils/addresses.js (hunk 3 above)
Comment (naddison36) [reply contracts/utils/addresses.js]: I think we can remove these. I had forgotten we can get the address from the deployment files.

File: contracts/utils/resolvers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (naddison36) [reply contracts/utils/resolvers.js]: I think this is ok as the `Proxy` is second in the `addresses.js` file. eg
```js
addresses.mainnet.OETHProxy = "0x856c4Efb76C1D1AE02e20CEB03A2A6a08b0b8dC3";
addresses.mainnet.WOETHProxy = "0xDcEe70654261AF21C44c093C300eD3Bb97b78192";
```

[the remark this answers is not in the document]
Comment (sparrowDom) [reply contracts/utils/resolvers.js]: Does the order in which the addresses are listed in the `address.js`?

**Pull request merged**